File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        int totalprof = 0;

        for(int i=1 ; i<prices.size(); i++)
        {
            // book profit, since increasing curve
            if(prices[i]>prices[i-1])
            {
                totalprof+= prices[i]-prices[i-1];
            }

        }

        return totalprof;
        
    }
};

File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.cs
public class Solution {
    public int MaxProfit(int[] prices) {

        int maxprof = 0;

        for(int i=1 ; i<prices.Length; i++)
        {
            // book profit since increasing curve
            if(prices[i]>prices[i-1])
            {
                maxprof+= prices[i]-prices[i-1];
            }
        }
        
        return maxprof;   
    }
}